### PR TITLE
[FrameworkBundle] do not use deprecated mailer.logger_message_listener service

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -1987,6 +1987,9 @@ private function registerMailerConfiguration(array $config, ContainerBuilder $co
         $container->getDefinition('mailer.transports')->setArgument(0, $transports);
         $container->getDefinition('mailer.default_transport')->setArgument(0, current($transports));
 
+        $container->removeDefinition('mailer.logger_message_listener');
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:1990]: why removing a definition and adding a deprecated alias following it ? Why not creating the right deprecated alias from the beginning ?

Comment (xabbuh) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:1990]: The `mailer.message_logger_listener` service is not always defined (only when the profiler is enabled).

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php (hunk 1 above)
Comment (stof) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:1990]: but here, you still remove the `mailer.logger_message_listener` definition all the time to replace it by an alias. So I don't see how this matters.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**